### PR TITLE
Misc decomp

### PR DESCRIPTION
**Pull request by ethteck**
Did some decomp of various early-rom files. Also I renamed the override flags variable accordingly and added macros for it since I'm almost positive they exist at this point.

### CODE REVIEW

File: src/code_106EF0.c [hunk 1]
```diff
@@ -29,32 +26,23 @@ s32 func_802E56D0(Entity* entity) {
...
-
-            entityIndex = create_entity(&D_802EA934, entity->position.x, entity->position.y, entity->position.z, 0.0f, 0x80000000);
-            temp_s0->unk_12 = entityIndex;
-            someEntity = get_entity_by_index(entityIndex);
-            temp2 = someEntity->dataBuf;
-            temp2->unk_00 = entity->listIndex;
-        }
+    struct802E3650* temp_s0 = entity->dataBuf;
+
+    if (temp_s0->unk_0A != 0xFFFF && get_global_flag(temp_s0->unk_0A) == 0) {
```
Comment (nanaian) [inline src/code_106EF0.c:32]: The RHS of the flag check should probably be a bool FALSE or just use `!`

File: src/code_1b40_len_20b0.c [hunk 2]
```diff
@@ -34,9 +35,155 @@ Gfx D_80074230[] = {
...
+        s32* overrideFlags = &gOverrideFlags;
+
+        if (*overrideFlags & 0x100) {
+            *overrideFlags |= 0x1000;
+        } else {
+            *overrideFlags &= ~0x1000;
+        }
+    }
+
+    {
```
Comment (nanaian) [inline src/code_1b40_len_20b0.c:135]: No way, Ethan using anonymous blocks!?

File: src/code_1b40_len_20b0.c [hunk 3]
```diff
@@ -34,9 +35,155 @@ Gfx D_80074230[] = {
...
+    *gDisplayContextPtr = &dispContexts[gCurrentDisplayContextIndex];
+    *gfx = &(*gDisplayContextPtr)->backgroundGfx[0];
+
+    gfx_init_state();
+    gfx_draw_background();
+
+    gDPFullSync((*gfx)++);
+    gSPEndDisplayList((*gfx)++);
+
+    ASSERT((s32)((u32)((*gfx) - (*gDisplayContextPtr)->backgroundGfx) << 3 >> 3) < ARRAY_COUNT((*gDisplayContextPtr)->backgroundGfx));
```
Comment (nanaian) [inline src/code_1b40_len_20b0.c:182]: what

File: src/code_43F0.c [hunk 4]
```diff
@@ -193,31 +218,27 @@ s32 func_80029994(s32 arg0) {
...
-    }
+    arg0 = abs(arg0);
 
-    ret = 0;
     if (arg0 != 0) {
-        if (arg0 == 1) {
-            return (func_80029994(1000) < 501) ^ 1;
+        switch (arg0) {
+            case 1:
+                ret = (func_80029994(1000) < 501) == 0;
```
Comment (nanaian) [inline src/code_43F0.c:229]: Use `!`?

File: src/code_43F0.c [hunk 5]
```diff
@@ -242,7 +263,32 @@ s32 round(f32 arg0) {
...
+    }
+
+    do {
+        if (theta < 0.0f) {
+            theta += 360.0f;
+        }
+        if (theta >= 360.0f) {
+            theta -= 360.0f;
+        }
+    } while (!(theta >= 0.0f) || !(theta < 360.0f));
```
Comment (nanaian) [inline src/code_43F0.c:288]: Does this match if you remove the `!`s and use `&&` instead

File: src/code_1b40_len_20b0.c [hunk 6]
```diff
@@ -34,9 +35,155 @@ Gfx D_80074230[] = {
...
+    step_current_game_mode();
+    update_entities();
+    func_80138198();
+    func_8014A548();
+    update_ambient_sounds();
+    func_80149734();
+    update_windows();
+    update_curtains();
+
+    overrideFlags = &gOverrideFlags;
```
Comment (nanaian) [inline src/code_1b40_len_20b0.c:89]: Given that you're using anonymous blocks in subsequent sections this should probably be in one too

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/code_1b40_len_20b0.c (hunk 3 above)
Comment (ethteck) [reply src/code_1b40_len_20b0.c:182]: https://discord.com/channels/279322074412089344/704241951972524063/822051304314241034

gonna leave a note but yeah this is a workaround to something that is not quite right

File: src/code_43F0.c (hunk 4 above)
Comment (ethteck) [reply src/code_43F0.c:229]: Thanks, I re-looked at this and realized I can do

```
            case 1:
                ret = func_80029994(1000) > 500;
                break;
```
which I think is a lot better

File: src/code_43F0.c (hunk 5 above)
Comment (ethteck) [reply src/code_43F0.c:288]: nary

**Review by nanaian — APPROVED** [review]

**Pull request merged**